DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -72,6 +74,23 @@ public List<(int id, string value)> GetCustomPromptsAlphabeticalList()
             return customPromptsDataService.GetCustomPromptsAlphabetical().ToList();
         }
 
+        public void AddCustomPromptToCentre(int centreId, int promptId, bool mandatory, string? options)
+        {
+            var centreCustomPrompts = GetCustomPromptsForCentreByCentreId(centreId);
+            var existingPromptNumbers = centreCustomPrompts.CustomPrompts
+                .Select(c => c.CustomPromptNumber)
+                .OrderBy(n => n);
+
+            var promptNumbers = new List<int> { 1, 2, 3, 4, 5, 6 };
+            promptNumbers.RemoveAll(x => existingPromptNumbers.Contains(x));
+            var promptNumber = promptNumbers.Any() ? promptNumbers.Min() : (int?)null;
+
+            if (promptNumber != null)
+            {
+                customPromptsDataService.AddCustomPromptToCentre(centreId, promptNumber.Value, promptId, mandatory, options);
+            }

```

Comment:
We might want to handle a failure with a warning log? And maybe redirect the user to the "Unexpected error" page?

Reply:
Agreed on the logging. The situations that this would occur seems unlikely but I guess it is better to handle it.